index.html

```diff
@@ -745,19 +745,30 @@ <h2>
         A <dfn>navigation scope</dfn> is a [[!URL]] that represents the set of
         URLs to which an <a>application context</a> can be navigated while the
         manifest is <a>applied</a>. The <a>navigation scope</a> of a manifest
-        <var>manifest</var> is <var>manifest</var>["<a>scope</a>"]. If this is
-        <code>undefined</code> (due to an error, or because it was not declared
-        in the manifest), the navigation scope is <dfn>unbounded</dfn>.
+        <var>manifest</var> is <var>manifest</var>["<a>scope</a>"].
       </p>
+      <div class="note" data-link-for="WebAppManifest">
+        <p>
+          If the <a>scope</a> member is not present in the manifest, it
+          defaults to the parent path of the <a>start_url</a> member. For
+          example, if <a>start_url</a> is <code>/pages/welcome.html</code>, and
+          <a>scope</a> is missing, the navigation scope will be
+          <code>/pages/</code> on the same origin. If <a>start_url</a> is
+          <code>/pages/</code> (the trailing slash is important!), the
+          navigation scope will be <code>/pages/</code>.
```

Comment:
So basically something like a base path or so. Isn't this defined somewhere?

If not, why not define it and create an algorithm, basically:

`base = (url) => url.substr(0, url.lastIndexOf('/') + 1)`

Reply:
Also what happens with `/hi/there/../cool/youcouldmakeit/'

Reply:
> So basically something like a base path or so. Isn't this defined somewhere?

Yes, in the [URL standard](https://url.spec.whatwg.org/#url-parsing). This section is a non-normative explanation with examples, for readers who want to understand practically what the default scope will be, without going into the URL parsing algorithm (which is practically unreadable).

The real (normative) text is in the ["scope" member](https://pr-preview.s3.amazonaws.com/mgiuca/manifest/pull/647.html#scope-member) section, where the default is defined as "a new URL using "." as input and *start URL* as the *base URL*". That invokes the [URL parser](https://url.spec.whatwg.org/#url-parsing) which then uses the standard resolution algorithm, the same as if you had `<a href=".">` to link to the containing directory of the current page. I did that because I don't think duplicating the resolution algorithm is a good idea, but it's a little arcane, hence the non-normative note.

If you want to explicitly define the logic here, instead of using the existing URL resolution algorithm, you need to:

1. Drop the query and fragment.
2. Drop everything after the last slash in the path.

The algorithm needs to operate on a URL object (not a string), so the path is represented as a list of segments, implicitly slash-separated. You need to be careful that the path ends with a blank segment, to represent a trailing slash. The algorithm would be this:

> 1. Let default be a new URL.
> 2. Set default’s username to start URL's username, default’s password to start URL's password, default’s host to start URL's host, default’s port to start URL's port, default’s path to a copy of start URL's path.
> 3. Remove default’s path’s last item, if any.
> 4. Append the empty string to default's path.

(This is basically a specialization of the URL Parser with *input* hard-coded as ".").

But I would rather not duplicate the existing logic.

(Aside: One of my main gripes about the URL standard is that almost the entire thing is a monolithic token-consume parser for the entire URL, rather than being broken into distinct concepts that you can easily point to, like a "resolve" algorithm which is what we need here. Formally, there is no such thing as a "relative URL" and there is no such thing as "resolution"; it's all just built-in functionality of the URL parser, which makes it hard to reference from outside, as we need to do here.)

Reply:
I wonder if @annevk has some comments for this?

Reply:
The language about "new URL" isn't quite correct. You actually need to invoke the URL parser. Otherwise I tend to agree with @mgiuca on the approach.

Reply:
I agree with @annevk but I was trying to be consistent with the writing in the rest of this document (in fact on the very next line: "Let scope URL be a new URL using value as input and manifest URL as the base URL."). There are six existing instances of this "new URL using X as input" in the Manifest spec.

I'll do a pass over it afterwards and fix up the invocations of URL parsing, but I think it's more urgent to land this to keep implementation divergence from getting out of control.

Reply:
I am fine with this, do you want me to land this now?